src/dnx/pal.win32.cpp

```diff
@@ -81,19 +83,56 @@ BOOL GetFullPath(LPCTSTR szPath, LPTSTR pszNormalizedPath)
     return TRUE;
 }
 
+namespace
+{
+    std::wstring get_runtime_path(TraceWriter& trace_writer)
+    {
+        wchar_t program_files_path_buffer[MAX_PATH];
+        wchar_t* program_files_paths[] =
+        {
+            L"ProgramFiles(x86)",
+#if !defined(_M_IX86)   // in case of 32-bit processes both ProgramFiles and ProgramFiles(x86) point to ProgramFiles(x86)
+            L"ProgramFiles"
+#endif
+        };
+
+        for (auto i = 0; i < sizeof(program_files_paths)/sizeof(wchar_t*); i++)
```

Comment:
Are you suggesting I should copy the code twice? (If so the answer is yes, I need a loop).